scripts/ci/crates.py

```diff
@@ -526,9 +527,22 @@ def __str__(self) -> str:
         return self.value
 
 
+VERSION_RE = re.compile(
+    "^(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?$"
+)
```

Comment:
NO! This is NOT acceptable code.

Use `re.VERBOSE` to split it up onto multiple lines and add comments to it.

Then add some unit tests that double as documentation

Reply:
I did not write this, it's the official semver regex. We use regular expressions all over the place. Why is this not acceptable?

I have no idea how I'm supposed to add unit tests to a python script, since we haven't done that before.

Reply:
https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string

Reply:
> I did not write this, it's the official semver regex

When you copy and paste code from the internet, ALWAYS include a reference to it.

> We use regular expressions all over the place. Why is this not acceptable?

Because it is 180 characters on one line with zero comments. Can you read it? I can't.

> I have no idea how I'm supposed to add unit tests to a python script, since we haven't done that before.

I don't know what the best practice is (ask in the `#python` channel on Slack?) but you could just add some asserts that are called from `main` (that's what I usually do)

Reply:
I replaced the regex with a basic check that the version string has exactly 3 period-separated digits.